sgkit/stats/pedigree.py

```diff
@@ -423,14 +423,24 @@ def _hamilton_kerr_self_kinship(
 
 @numba_jit
 def _hamilton_kerr_pair_kinship(
+    tau_p: int,
+    tau_q: int,
+    kinship_pj: float,
+    kinship_qj: float,
+) -> float:  # pragma: no cover
+    ploidy_i = tau_p + tau_q
+    return (tau_p / ploidy_i) * kinship_pj + (tau_q / ploidy_i) * kinship_qj
+
+
+@numba_jit
+def _insert_hamilton_kerr_pair_kinship(
     kinship: ArrayLike, parent: ArrayLike, tau: ArrayLike, i: int, j: int
 ) -> None:  # pragma: no cover
     p, q = parent[i, 0], parent[i, 1]
```

Comment:
Good point, there is certainly a perf penalty for implicit unpacking (`p, q = parent[i]`). Numba tuples are actually typed with fixed length so it may be zero cost but I'll check the difference.